lib/cloudinary/api.rb

```diff
@@ -30,8 +30,30 @@ def self.ping(options={})
     call_api(:get, "ping", {}, options)
   end
 
+  # Gets account usage details
+  #
+  # Get a report on the status of your Cloudinary account usage details, including
+  # storage, bandwidth, requests, number of resources, and add-on usage.
+  # Note that numbers are updated periodically.
+  #
+  # @see https://cloudinary.com/documentation/admin_api#get_account_usage_details Get account usage details
+  #
+  # @param [Hash] options Additional options
+  # @return [Cloudinary::Api::Response]
+  # @raise [Cloudinary::Api:Error]
   def self.usage(options={})
-    call_api(:get, "usage", {}, options)
+    uri = 'usage'
+    date = options[:date]
+
+    unless date.nil?
+      if date.is_a?(Date)
```

Comment:
It would make more sense to move type check to `to_usage_api_date_format`. So here would only be:
```
uri += "/#{Cloudinary::Utils.to_usage_api_date_format(date)}" unless date.nil?
```